lucille-plugins/lucille-weaviate/src/test/java/com/kmwllc/lucille/weaviate/indexer/WeaviateIndexerTest.java

```diff
@@ -0,0 +1,191 @@
+    for (int i = 1; i <= events.size(); i++) {
+      Assert.assertEquals("doc" + i, events.get(i - 1).getDocumentId());
+      Assert.assertEquals(Event.Type.FINISH, events.get(i - 1).getType());
+    }
+  }
+
+  @Test
+  public void testWeaviateIndexerException() throws Exception {
+    PersistingLocalMessageManager manager = new PersistingLocalMessageManager();
+    Config config = ConfigFactory.load("WeaviateIndexerTest/config.conf");
+
+    Document doc = Document.create("doc1", "test_run");
+    Document doc2 = Document.create("doc2", "test_run");
+    Document doc3 = Document.create("doc3", "test_run");
+    Document doc4 = Document.create("doc4", "test_run");
+    Document doc5 = Document.create("doc5", "test_run");
+
+    WeaviateIndexer indexer = new CorruptedWeaviateIndexer(config, manager, mockClient, "testing");
+    manager.sendCompleted(doc);
+    manager.sendCompleted(doc2);
+    manager.sendCompleted(doc3);
+    manager.sendCompleted(doc4);
+    manager.sendCompleted(doc5);
+    indexer.run(5);
+
+    List<Event> events = manager.getSavedEvents();
+    Assert.assertEquals(5, events.size());
+    for (int i = 1; i <= events.size(); i++) {
+      Assert.assertEquals("doc" + i, events.get(i - 1).getDocumentId());
+      Assert.assertEquals(Event.Type.FAIL, events.get(i - 1).getType());
+    }
+  }
+
+  @Test
+  public void testValidateConnection() {
+    PersistingLocalMessageManager manager = new PersistingLocalMessageManager();
+    Config config = ConfigFactory.load("WeaviateIndexerTest/config.conf");
+    WeaviateIndexer indexer = new WeaviateIndexer(config, manager, mockClient, "testing");
+    Assert.assertTrue(indexer.validateConnection()); // should only work the first time with the mockClient
+    Assert.assertFalse(indexer.validateConnection());
+    Assert.assertFalse(indexer.validateConnection());
+  }
+
+  @Test
+  public void testMultipleBatches() throws Exception {
+    PersistingLocalMessageManager manager = new PersistingLocalMessageManager();
+    Config config = ConfigFactory.load("ElasticsearchIndexerTest/batching.conf");
+    WeaviateIndexer indexer = new WeaviateIndexer(config, manager, mockClient, "testing");
+
+    Document doc = Document.create("doc1", "test_run");
+    Document doc2 = Document.create("doc2", "test_run");
+    Document doc3 = Document.create("doc3", "test_run");
+    Document doc4 = Document.create("doc4", "test_run");
+    Document doc5 = Document.create("doc5", "test_run");
+
+    manager.sendCompleted(doc);
+    manager.sendCompleted(doc2);
+    manager.sendCompleted(doc3);
+    manager.sendCompleted(doc4);
+    manager.sendCompleted(doc5);
+    indexer.run(5);
+
+    ArgumentCaptor<WeaviateObject> bulkRequestArgumentCaptor = ArgumentCaptor.forClass(WeaviateObject.class);
+
+    // todo should i be capturing a different method to test that there were multiple batches?
+    verify(mockObjectsBatcher, times(5)).withObject(bulkRequestArgumentCaptor.capture());
```

Comment:
Which part of the WeaviateIndexer should be tested here that is consistent with calling batch size=2 three times?

Reply:
This is the line in WeaviateIndexer that sends a batch:

`      Result<ObjectGetResponse[]> result = batcher.withConsistencyLevel(ConsistencyLevel.ALL).run();`

So ideally we'd like to test that run() is called three times (i.e. once for each of the three batches).